src/renderer/widgets/AssetTransactionModal/ui/AssetTransactionModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { useUnit } from 'effector-react';
+import { Link } from 'react-router-dom';
+
+import { useI18n } from '@app/providers';
+import { BaseModal, BodyText, FootnoteText, HeaderTitleText, Icon, SearchInput } from '@shared/ui';
+import { useModalClose } from '@shared/lib/hooks';
+import { totalAmount } from '@shared/lib/utils';
+import { PathType, Paths, createLink } from '@shared/routes';
+import { ChainIcon } from '@entities/chain';
+import { networkModel } from '@entities/network';
+import { AssetFiatBalance } from '@entities/price';
+import { assetTransactionModel } from '../model/asset-transaction-model';
+import { assetTransactionUtils } from '../lib/utils';
+import { ModalType } from '../lib/types';
+
+type ModalDetailsProps = { title: string; path: PathType };
+
+const getModalDetails = (modalType: ModalType): ModalDetailsProps => {
+  const modalDetailsMap = {
+    [ModalType.TRANSFER]: {
+      title: 'operations.modalTitles.transferOn',
+      path: Paths.TRANSFER_ASSET,
+    },
+    [ModalType.RECEIVE]: {
+      title: 'receive.title',
+      path: Paths.RECEIVE_ASSET,
+    },
+  };
+
+  return modalDetailsMap[modalType];
+};
+
+export const AssetTransactionModal = () => {
+  const { t } = useI18n();
+
+  const assetWithChains = useUnit(assetTransactionModel.$assetWithChains);
+  const step = useUnit(assetTransactionModel.$step);
+  const modalType = useUnit(assetTransactionModel.$modalType);
+  const query = useUnit(assetTransactionModel.$query);
+  const chains = useUnit(networkModel.$chains);
+
+  const [isModalOpen, closeModal] = useModalClose(
+    !assetTransactionUtils.isNoneStep(step),
+    assetTransactionModel.output.flowClosed,
+  );
+
+  if (!assetWithChains || modalType === null) return null;
+
+  const path = getModalDetails(modalType).path;
+  const title = (
+    <HeaderTitleText>{t(getModalDetails(modalType).title, { asset: assetWithChains.symbol })}</HeaderTitleText>
+  );
+
+  return (
+    <BaseModal
+      closeButton
+      panelClass="max-h-[610px] overflow-y-auto"
+      headerClass="p-3 pl-5 pb-7"
+      isOpen={isModalOpen}
+      title={title}
+      onClose={closeModal}
+    >
+      <SearchInput
+        value={query}
+        placeholder={t('balances.searchPlaceholder')}
+        className="w-full"
+        onChange={assetTransactionModel.events.queryChanged}
+      />
+      <FootnoteText className="text-text-tertiary pt-4 pb-2">{t('portfolilo.selectNetworkLabel')}</FootnoteText>
+      <ul>
+        {assetWithChains.chains.map((chain) => (
+          <li
+            key={`${chain.assetSymbol}_${chain.chainId}`}
+            role="button"
```

Comment:
Does it actually need that button role? When `Link` has everything it needs in terms of a11y